cargo-dist/src/tasks.rs

```diff
@@ -1400,6 +1400,11 @@ impl<'pkg_graph> DistGraphBuilder<'pkg_graph> {
         }
     }
 
+    fn add_cyclonedx_sbom_file(&mut self, to_release: ReleaseIdx) {
+        // FIXME: Figure out the location
+        eprintln!("add_cyclonedx_sbom_file()");
```

Comment:
You mentioned this earlier, but looks like it places the output in a file named `packagename/packagename.cdx.xml`.

The relationship between release/package/etc. is kind of thorny here, so let me outline the shape of what you need here. You're going to want this to have  `to_package: PackageIdx` in addition to  `to_release: ReleaseIdx` (I'll get into why the release is necessary later). You can get that index from `info.package_idx` within `compute_releases`, like on this line: https://github.com/axodotdev/cargo-dist/blob/f8e0688ce296ed262d6d4c63c676f62fb5e3e48b/cargo-dist/src/tasks.rs#L2629

With access to that, you can `let package = self.workspaces.package(package_idx)`. That gives you an `axoproject::PackageInfo`: https://github.com/axodotdev/cargo-dist/blob/f8e0688ce296ed262d6d4c63c676f62fb5e3e48b/axoproject/src/lib.rs#L404

Which itself has the `true_name` and `package_root` fields you need to construct the artifact path. (The `true_name` here is because the name can be overridden in dist config, but we're talking to Cargo and so we need the genuine Cargo name.)

The method itself can then access config that can conditionally enable or disable itself. We'll need to grab the release so we can get the release-specific/package-scoped config (since I believe you've set the cyclonedx stuff to be something you can enable/disable per-package), then return if it's disabled.

```rust
fn add_cyclonedx_sbom_file(&mut self, to_package: PackageIdx, to_release: ReleaseIdx) {
  let release = self.release(to_release);

  if !self.global_artifacts_enabled() || !release.config.builds.cargo.cargo_cyclonedx {
    return
 }

  // From here you can safely proceed with constructing the artifact
}
```